yorkie/src/main/kotlin/dev/yorkie/document/crdt/RgaTreeSplit.kt

```diff
@@ -0,0 +1,575 @@
+ * the block is split.
+ */
+@Suppress("UNCHECKED_CAST")
+internal class RgaTreeSplit<T : CharSequence> : Iterable<RgaTreeSplitNode<T>> {
+    val head = RgaTreeSplitNode(InitialNodeID, INITIAL_NODE_VALUE as T)
+    private val treeByIndex =
+        SplayTreeSet<RgaTreeSplitNode<T>> { it.contentLength }.apply { insert(head) }
+    private val treeByID = TreeMap<RgaTreeSplitNodeID, RgaTreeSplitNode<T>>().apply {
+        put(head.id, head)
+    }
+    private val removedNodeMap = mutableMapOf<RgaTreeSplitNodeID, RgaTreeSplitNode<T>>()
+
+    val length
+        get() = treeByIndex.length
+
+    val removedNodesLength
+        get() = removedNodeMap.size
+
+    /**
+     * Does following stpes.
+     * 1. Split nodes with the given [range].
+     * 2. Delete between the given [range].
+     * 3. Insert a new node.
+     * 4. Add removed nodes.
+     */
+    fun edit(
+        range: RgaTreeSplitNodeRange,
+        executedAt: TimeTicket,
+        value: T?,
+        latestCreatedAtMapByActor: Map<ActorID, TimeTicket>? = null,
+    ): Triple<RgaTreeSplitNodePos, Map<ActorID, TimeTicket>, List<TextChange>> {
+        // 1. Split nodes.
+        val (toLeft, toRight) = findNodeWithSplit(range.second, executedAt)
+        val (fromLeft, fromRight) = findNodeWithSplit(range.first, executedAt)
+
+        // 2. Delete between from and to.
+        val nodesToDelete = if (toRight != null && fromRight != null) {
+            findBetween(fromRight, toRight)
+        } else {
+            emptyList()
+        }
+
+        val (changes, latestCreatedAtMap, removedNodeMapByNodeKey) = deleteNodes(
+            nodesToDelete,
+            executedAt,
+            latestCreatedAtMapByActor,
+        )
+        val caretID = toRight?.id ?: toLeft.id
+        var caretPos = RgaTreeSplitNodePos(caretID, 0)
+
+        // 3. Insert a new node.
+        value?.let {
+            val index = findIndexFromNodePos(fromLeft.createRange().second, true)
+            val inserted = insertAfter(
+                fromLeft,
+                RgaTreeSplitNode(
+                    RgaTreeSplitNodeID(executedAt, 0),
+                    it,
+                ),
+            )
+            if (changes.isNotEmpty() && changes.last().from == index) {
+                changes.last().content = it.toString()
+            } else {
+                changes.toMutableList().add(
+                    TextChange(
+                        TextChangeType.Content,
+                        executedAt.actorID,
+                        index,
+                        index,
+                        value.toString(),
+                    ),
+                )
+            }
+            caretPos = RgaTreeSplitNodePos(inserted.id, inserted.contentLength)
+        }
+
+        // 4. Add removed nodes.
+        removedNodeMapByNodeKey.forEach {
+            removedNodeMap[it.key] = it.value
+        }
+
+        return Triple(caretPos, latestCreatedAtMap, changes)
```

Comment:
added TextChange element would be lost as we are passing the instance created by `toMutableList`